apps/files/src/components/Collaborators/NewCollaborator.vue

```diff
@@ -169,7 +169,10 @@ export default {
             )
 
             const exists = this.shares.find(existingCollaborator => {
-              return collaborator.value.shareWith === existingCollaborator.name
+              return (
+                collaborator.value.shareWith === existingCollaborator.name &&
+                parseInt(collaborator.value.shareType) === parseInt(existingCollaborator.info.share_type)
```

Comment:
in general always use `parseInt(x, 10)` to be on the safe side

Reply:
Right - fixed